lsp/nls/src/server.rs

```diff
@@ -70,7 +76,34 @@ impl Server {
             cache,
             lin_cache,
             initial_ctxt,
+            initial_env: Environment::new(),
+        }
+    }
+
+    pub fn initialize_stdlib_environment(&mut self) -> Option<()> {
+        let mut initial = Environment::new();
+        let modules = stdlib::modules();
+        for module in modules {
+            // This module has a different format from the rest of the stdlib items
+            // Also, users are not supposed to use the internal module directly
+            if module == StdlibModule::Internals {
+                continue;
+            }
+            let name: Ident = module.into();
+            let file_id = self.cache.get_submodule_file_id(module)?;
+            let lin = self.lin_cache_get(&file_id).ok()?;
+            // We're using the ID 0 here because the stdlib is represented by a
+            // single record which would have an ID of 0
```

Comment:
```suggestion
            // We're using the ID 0 here because a stdlib module is currently
            // REQUIRED to be a record literal. The linearization ID of this
            // record must thus be 0
```